File: PythonSelenium/PythonApplication1/src/pageobjects/NavBar.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


class NavBar():
    def navigate_to_login(driver):
        driver.find_element(By.CSS_SELECTOR, "button.cookieMessage__button.btn.btn-primary").click()
        driver.find_element(By.CSS_SELECTOR, "a.ico-user.paws-ico.paws-loaded").click()
        driver.find_element(By.ID, "signin").click()

    def navigate_to_favoriates(driver):
        driver.find_element(By.CLASS_NAME, "ico-favourite").click()

    def navigate_to_basket(driver):
        driver.find_element(By.CLASS_NAME, "ico-basket").click()

    def add_text_to_search(driver, search_text):
        driver.find_element(By.CLASS_NAME, "search-input").clear()
        driver.find_element(By.CLASS_NAME, "search-input").send_keys(search_text)
        element = WebDriverWait(driver, 3).until(
            EC.presence_of_element_located((By.XPATH, "//*[@class='product-matches'][contains(@style, 'display: block;')]"))
        )


        

File: Seleniulcsharp/Seleniulcsharp/src/pageobjects/NavBar.cs
using OpenQA.Selenium;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using OpenQA.Selenium.Support.UI;
using SeleniumExtras.WaitHelpers;

namespace Seleniulcsharp.src.pageobjects
{
    public class NavBar
    {
        public void navigate_to_login(IWebDriver driver)
        {
            driver.FindElement(By.CssSelector("button.cookieMessage__button.btn.btn-primary")).Click();
            driver.FindElement(By.CssSelector("a.ico-user.paws-ico.paws-loaded")).Click();
            driver.FindElement(By.Id("signin")).Click();
        }

        public void navigate_to_favoriates(IWebDriver driver)
        {
            driver.FindElement(By.ClassName("ico-favourite")).Click();
        }

        public void navigate_to_basket(IWebDriver driver)
        {
            driver.FindElement(By.ClassName("ico-basket")).Click();
        }

        public void add_text_to_search(IWebDriver driver, string search_text)
        {
            driver.FindElement(By.ClassName("search-input")).Clear();
            driver.FindElement(By.ClassName("search-input")).SendKeys(search_text);
            OpenQA.Selenium.Support.UI.WebDriverWait

            wait = new WebDriverWait(driver, TimeSpan.FromSeconds(5));
            wait.Until(ExpectedConditions.PresenceOfAllElementsLocatedBy(By.XPath("//*[@class='product-matches'][contains(@style, 'display: block;')]")));
        }
    }
}
